jgrapht-core/src/test/java/org/jgrapht/alg/KDisjointShortestPathsTest.java

```diff
@@ -0,0 +1,317 @@
+        DefaultWeightedEdge e25 = graph.addEdge(2, 5);
+        DefaultWeightedEdge e13 = graph.addEdge(1, 3);
+        DefaultWeightedEdge e35 = graph.addEdge(3, 5);
+        DefaultWeightedEdge e14 = graph.addEdge(1, 4);
+        DefaultWeightedEdge e45 = graph.addEdge(4, 5);
+        DefaultWeightedEdge e23 = graph.addEdge(2, 3);
+        DefaultWeightedEdge e34 = graph.addEdge(3, 4);
+        
+        graph.setEdgeWeight(e12, 1);
+        graph.setEdgeWeight(e25, 6);
+        graph.setEdgeWeight(e13, 4);
+        graph.setEdgeWeight(e35, 5);
+        graph.setEdgeWeight(e14, 4);
+        graph.setEdgeWeight(e45, 1);
+        graph.setEdgeWeight(e23, 1);
+        graph.setEdgeWeight(e34, 1);
+        
+        KDisjointShortestPaths<Integer, DefaultWeightedEdge> alg = new KDisjointShortestPaths<>(graph, 1, 5);
+        
+        List<GraphPath<Integer, DefaultWeightedEdge>> pathList = alg.getPaths(5);
+        assertEquals(3, pathList.size());
+        assertEquals(2, pathList.get(0).getLength());
+        assertEquals(2, pathList.get(1).getLength());
+        assertEquals(2, pathList.get(2).getLength());
+        
+        assertEquals(5.0, pathList.get(0).getWeight());
+        assertEquals(7.0, pathList.get(1).getWeight());
+        assertEquals(9.0, pathList.get(2).getWeight());
+        
+        assertTrue(pathList.get(0).getEdgeList().contains(e14));
+        assertTrue(pathList.get(0).getEdgeList().contains(e45));
+        
+        assertTrue(pathList.get(1).getEdgeList().contains(e12));
+        assertTrue(pathList.get(1).getEdgeList().contains(e25));                       
+        
+        assertTrue(pathList.get(2).getEdgeList().contains(e13));
+        assertTrue(pathList.get(2).getEdgeList().contains(e35));
+    }
+    
+    /**
+     * Only single disjoint path should exist on the line
+     */
+    public void testLinear() {
+        Graph<Integer, DefaultWeightedEdge> graph = new DefaultDirectedGraph<>(DefaultWeightedEdge.class);  
+        GraphGenerator<Integer, DefaultWeightedEdge, Integer> graphGenerator = new LinearGraphGenerator<>(20);
+        graphGenerator.generateGraph(graph, new VertexFactory<Integer>() {
+            
+            private int i = 1;
+            
+            @Override
+            public Integer createVertex() {
+                return i++;
+            }
+        }, null);
+        
+        KDisjointShortestPaths<Integer, DefaultWeightedEdge> alg = new KDisjointShortestPaths<>(graph, 1, 2);
+        List<GraphPath<Integer, DefaultWeightedEdge>> pathList = alg.getPaths(20);
+        
+        assertEquals(1, pathList.size());
+        assertEquals(19, pathList.get(0).getLength());
+        assertEquals(19.0, pathList.get(0).getWeight());
+        
+        for (int i = 1; i < 21; i++) {
+            assertTrue(pathList.get(0).getVertexList().contains(i));
+        }
+    }
+    
+    /**
+     * Exactly single disjoint path should exist on the ring
+     */
+    public void testRing() {

```

Comment:
This test is not really useful. You implement an algorithm to compute k-edge-disjoint shortest paths. Moreover, the sum of costs of these k paths has to be minimal. However, in this test, you create a graph which only contains 1 path. Because you only return 1 path, you cannot test whether the resulting paths are indeed edge disjoint. Moreover, you cannot test whether specifically these paths are of minimal total weight. Same for the 'linear' test.

Reply:
The KDisjointShortestPaths algorithm is first a shortest path algorithm, then K-shortest and then K-disjoint-shortest. So, I find these basic graph tests useful at least as sanity tests. It proved to be so during my work on the algorithm. I strongly recommend to keep it.